src/app.rs

```
@@ -142,14 +142,24 @@ pub struct ApplicationBuilder<'a, 'b, T: State + 'a> {
 impl<'a, 'b, T: State + 'a> ApplicationBuilder<'a, 'b, T> {
     /// Creates a new ApplicationBuilder with the given initial game state and
     /// display configuration.
-    pub fn new(initial_state: T) -> Self {
-        ApplicationBuilder {
-            base_path: format!("{}/resources", env!("CARGO_MANIFEST_DIR")).into(),
+    pub fn new(initial_state: T) -> Result<Self> {
+        use num_cpus;
+        use rayon::Configuration;
+
+        let num_cores = num_cpus::get();
+        let cfg = Configuration::new().num_threads(num_cores);
+        let pool = ThreadPool::new(cfg).map(|p| Arc::new(p)).map_err(|_| Error::Application)?;
```

We shouldn't discard errors like this.